### PR TITLE
Fix #212

### LINKED ISSUE #212: Style Changelog Page
- Add icons for headers (i.e. Bug fixes, Breaking changes, etc)

- adjust list/bullet points

- Improve layout/type/color use for readability

### PR DESCRIPTION
**Pull request by ststimac**
Create changelog template with updated styles

### CODE REVIEW

File: src/hexo/themes/documentation/source/images/breaking-changes-icon.svg [hunk 1]
```diff
@@ -0,0 +1 @@
+<svg id="Layer_1" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 34.9 34.9"><style>.st0{fill:url(#background_1_);} .st1{fill:#FFFFFF;} .st2{fill:none;} .st3{fill:#4D4D4D;} .st4{fill:#4046DD;} .st5{fill:#231F20;} .st6{fill:none;stroke:#4046DD;stroke-width:2;stroke-miterlimit:10;}</style><circle class="st4" cx="17.1" cy="17.4" r="16.1"/><path class="st1" d="M19 22.5c.4.4.6.9.6 1.6 0 .6-.2 1.2-.6 1.6-.4.4-.9.6-1.6.6-.6 0-1.1-.2-1.5-.6-.4-.4-.6-.9-.6-1.6 0-.6.2-1.2.6-1.6.4-.4.9-.6 1.5-.6s1.2.2 1.6.6zm.6-13.9v4.5l-.8 6.9h-2.7l-.8-6.9V8.6h4.3z"/></svg>
```
Comment (alrra) [inline src/hexo/themes/documentation/source/images/breaking-changes-icon.svg]:
```svg
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 34.9 34.9"><circle fill="#4046dd" cx="17.1" cy="17.4" r="16.1"/><path fill="#fff" d="M19 22.5c.4.4.6.9.6 1.6 0 .6-.2 1.2-.6 1.6-.4.4-.9.6-1.6.6-.6 0-1.1-.2-1.5-.6-.4-.4-.6-.9-.6-1.6 0-.6.2-1.2.6-1.6.4-.4.9-.6 1.5-.6s1.2.2 1.6.6zm.6-13.9v4.5l-.8 6.9h-2.7l-.8-6.9V8.6h4.3z"/></svg>
```

File: src/hexo/themes/documentation/source/images/bug-fixes-icon.svg [hunk 2]
```diff
@@ -0,0 +1 @@
+<svg id="Layer_1" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 34.9 34.9"><style>.st0{fill:url(#background_1_);} .st1{fill:#FFFFFF;} .st2{fill:none;} .st3{fill:#4D4D4D;} .st4{fill:#4046DD;} .st5{fill:#231F20;} .st6{fill:none;stroke:#4046DD;stroke-width:2;stroke-miterlimit:10;}</style><circle class="st4" cx="17.4" cy="17.4" r="16.1"/><path class="st1" d="M20.66587 13.88637l3.54855-2.05369.60108 1.0386-3.54855 2.05369zM20.51251 23.05691l.77148-.9192 3.1406 2.63589-.77148.9192zM10.33274 12.87172l.60096-1.0386 3.54855 2.05328-.60096 1.0386zM10.7036 24.82122l3.14019-2.63589.77148.91908-3.14019 2.63589zM9 18h3.1v1.2H9zM22.8 18h3.1v1.2h-3.1zM17.6 9.1c-1.5 0-2.8 1-3.1 2.5h6.3c-.4-1.4-1.7-2.5-3.2-2.5z"/><path class="st1" d="M17.1 11.3c-.4 0-.8.1-1.1.2h1.1v-.2zM11.9 18.5c0 3.9 2.3 7 5.2 7.2V12.6h-2.8c-1.4 1.3-2.4 3.5-2.4 5.9zM18.1 11.3v.2h1.1c-.3 0-.7-.1-1.1-.2zM20.9 12.6h-2.8v13.1c2.9-.2 5.2-3.3 5.2-7.2-.1-2.4-1-4.6-2.4-5.9z"/></svg>
```
Comment (alrra) [inline src/hexo/themes/documentation/source/images/bug-fixes-icon.svg]:
```svg
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 34.9 34.9"><style>.e{fill:#fff}.a{fill:#4046DD}</style><circle cx="17.4" cy="17.4" r="16.1" class="a"/><path d="M20.666 13.886l3.548-2.053.602 1.038-3.55 2.056zm-.153 9.17l.77-.918 3.142 2.636-.772.92zm-10.18-10.184l.6-1.04 3.55 2.054-.602 1.04zm.37 11.95l3.14-2.637.772.92-3.14 2.635zM9 18h3.1v1.2H9zm13.8 0h3.1v1.2h-3.1zm-5.2-8.9c-1.5 0-2.8 1-3.1 2.5h6.3c-.4-1.4-1.7-2.5-3.2-2.5z" class="e"/><path d="M17.1 11.3c-.4 0-.8.1-1.1.2h1.1v-.2zm-5.2 7.2c0 3.9 2.3 7 5.2 7.2V12.6h-2.8c-1.4 1.3-2.4 3.5-2.4 5.9zm6.2-7.2v.2h1.1c-.3 0-.7-.1-1.1-.2zm2.8 1.3h-2.8v13.1c2.9-.2 5.2-3.3 5.2-7.2-.1-2.4-1-4.6-2.4-5.9z" class="e"/></svg>
```

File: src/hexo/themes/documentation/source/images/new-features-icon.svg [hunk 3]
```diff
@@ -0,0 +1 @@
+<svg id="Layer_1" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 34.9 34.9"><style>.st0{fill:url(#background_1_);} .st1{fill:#FFFFFF;} .st2{fill:none;} .st3{fill:#4D4D4D;} .st4{fill:#4046DD;} .st5{fill:#231F20;} .st6{fill:none;stroke:#4046DD;stroke-width:2;stroke-miterlimit:10;}</style><circle class="st4" cx="17.4" cy="17.4" r="16.1"/><path class="st1" d="M13 7.3l1.4 2.7 2.7 1.4-2.7 1.5-1.4 2.6-1.5-2.6-2.6-1.5 2.6-1.4zM22 6.2L23 8l1.7.9-1.7 1-1 1.8-1-1.8-1.8-1L21 8zM19 14.7l2.5 4.5 4.5 2.5-4.5 2.5-2.5 4.5-2.5-4.5-4.5-2.5 4.5-2.5z"/></svg>
```
Comment (alrra) [inline src/hexo/themes/documentation/source/images/new-features-icon.svg]:
```svg
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 34.9 34.9"><circle cx="17.4" cy="17.4" r="16.1" fill="#4046dd"/><path fill="#fff" d="M13 7.3l1.4 2.7 2.7 1.4-2.7 1.5-1.4 2.6-1.5-2.6-2.6-1.5 2.6-1.4zm9-1.1L23 8l1.7.9-1.7 1-1 1.8-1-1.8-1.8-1L21 8zm-3 8.5l2.5 4.5 4.5 2.5-4.5 2.5-2.5 4.5-2.5-4.5-4.5-2.5 4.5-2.5z"/></svg>
```

**Comment by molant** [comment]
@alrra can you please update the svg files in this PR?

**Comment by alrra** [comment]
> can you please update the svg files in this PR?

[Done](https://github.com/ststimac/sonarwhal.com-1/commit/3fb2d1326ecd4139f8df14de789153da7d775bd4).

**Comment by qzhou1607** [comment]
@molant @alrra Logic was added. A `changelog.yml` will be generated in `_data` when running `updater` and thus used for populating the page.

I started off by extract details (commit sha, message text and PR id of associated commit) from changelog items in the markdown to directly interpolate the variables in the template.  But soon realized direct interpolation won't work for some messages that need rendering. I kept the extraction logic in the code, but it can be removed if preferred.

File: helpers/updater.js [hunk 4]
```diff
@@ -14,6 +17,18 @@ const divider = '---';
 
 console.log('Updater is initiated.');
 
+// Make sure the options used here are the same as used in `hexo`.
+marked.setOptions({
```
Comment (molant) [inline helpers/updater.js]: Can we have this options in a config file and use it in both places?

Comment (qzhou1607) [reply helpers/updater.js]: @molant Yes, just made the change.

**Pull request merged**